play/src/front/Chat/Connection/Matrix/tests/MatrixChatConnection.test.ts

```diff
@@ -239,37 +239,40 @@ describe("MatrixChatConnection", () => {
     });
 
     describe("startMatrixClient", () => {
-        it.each([[ClientEvent.Sync], [ClientEvent.Room], [ClientEvent.DeleteRoom], [RoomEvent.MyMembership]])(
-            "should call this.client.on for event %s",
-            async (expectedEventName) => {
-                const onMock = vi.fn();
-                const mockMatrixClient = {
-                    isGuest: vi.fn(),
-                    on: onMock,
-                    store: {
-                        startup: vi.fn(),
-                    },
-                    initRustCrypto: vi.fn(),
-                    startClient: vi.fn(),
-                } as unknown as MatrixClient;
+        it.each([
+            [ClientEvent.Sync],
+            [ClientEvent.Room],
+            [ClientEvent.DeleteRoom],
+            [RoomEvent.MyMembership],
+            [RoomEvent.Name],
+            ["RoomState.events"],
+        ])("should call this.client.on for event %s", async (expectedEventName) => {
```

Comment:
This whole test is probably too low level. You are testing implementation details (testing the fact that we are indeed subscribing to an event handler is an implementation detail).